File: PN25.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;       

    public:
        int iCnt = 0;
		
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        void DisplayF()      
        {
			 for(iCnt = 1; iCnt<= iNo; iCnt++)
             {
                 cout<<iCnt<<endl;        
             }
        }

        bool CheckPrime()      
        {
			int iFactCnt = 0;
			 for(iCnt = 2; iCnt <= (iNo/2); iCnt++)
             {
                 if(iNo % iCnt == 0)
				 {
					 break;
				 }					 
             }
			 if(iCnt == (iNo/2)+1)
			 {
				 return true;
			 }
			 else
			 {
				 return false;
			 }
			 
        }
		
		

};

int main()
{
    Number nobj;
	
	bool bRet;


    nobj.Accept();
    nobj.DisplayF();

    bRet = nobj.CheckPrime();
	
	if(bRet == true)
	{
		cout<<"it is prime number: "<<endl;
	}
	else
	{
		cout<<"it is not prime number: "<<endl;
	}


    return 0;
}

File: PN25.java
// Template for probplems on Numbers

import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }
    
	public int Reverse()
	{
     int iDigit = 0;
	 int iRev = 0;
	
	 if(iNo < 0)
	 {
		iNo = -iNo;
	 }
	 while(iNo > 0)
	 {
		iDigit = iNo % 10;
		iRev = iRev * 10 + iDigit;

		iNo = iNo/10;
	 }
	 return iRev;
	}
}

class PN25
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            int iRet = 0;
			
            nobj.Accept();
            nobj.Display();

            iRet = nobj.Reverse();
			
			System.out.println("Reverse number is : "+iRet);
    }
}